src/core/utils/Utilities.ts

```diff
@@ -170,3 +172,32 @@ export const decodeBlock = (input: any, inputOffset: number, output: any, output
     output[outputOffset + 3] = ((bytes[4] & 0x01) << 7) | (bytes[5] << 2) | (bytes[6] >> 3);
     output[outputOffset + 4] = ((bytes[6] & 0x07) << 5) | bytes[7];
 };
+
+/**
+ * Generate keccak 512 hash given by data.
+ * @param {Uint8Array} data The data to hash.
+ * @returns {Uint8Array} The keccak hash value.
+ * */
+export const keccakHash = (data: Uint8Array): Uint8Array => {
+    const hasher = new Keccak(512);
+    const hex = hasher.update(Buffer.from(data)).digest('hex');
+    return Converter.hexToUint8(hex);
```

Comment:
looking at https://www.npmjs.com/package/sha3 it seems that by default digest() can return binary,
so you'd avoid converting it to hex an back, wrapping it with Uint8Array should do the trick

Reply:
By default it return binary, yes,

Reply:
Alright, I will take a look at it.